src/WS2812FX/effects/Fire2012Effect.cpp

```diff
@@ -5,19 +5,23 @@
 bool Fire2012Effect::init(WS2812FX* strip) {
     // Allocate and initialize heat array for fire simulation
     if (!allocateHeatArray(strip)) {
+        setInitialized(false);
         return false; // Failed to allocate memory
     }
     
     // Clear heat array to start with a cold fire
     memset(heatArray, 0, heatArraySize * sizeof(byte));
     
     // Use standard initialization pattern from helper
-    return EffectHelper::standardInit(strip, timebase, initialized);
+    bool tempInit = false;
```

Comment:
see comment on src/WS2812FX/effects/BeatsinGlowEffect.cpp line R7

and I would prefer having standardInit before allocating any array or initialising other stuff.